### DISCUSSION TITLE
Is it Possible to Show Node IDs in the Diagram/Flowchart?

### DISCUSSION THREAD

**Discussion by Foxxsayy** [squidfunk/mkdocs-material#6439] [Q&A]
I'm always forgetting exactly what I typed for a node name and cumulatively spend a ton of time looking up the name of the node I want to connect to, especially when coming back to a diagram. Is it possible to either:

1. Enable the Node ID to show in a tooltip when a node is moused over, or
2. Display the Node ID on the node itself, like a label?

2 Visual Examples:
![image](https://github.com/squidfunk/mkdocs-material/assets/98929636/2e0b37ec-6534-4a4c-b6f2-808012203ec3)

![image](https://github.com/squidfunk/mkdocs-material/assets/98929636/da5c613d-38d4-4cb3-954a-6afc11e8ec6c)

**Comment by kamilkrzyskow** [comment]
Hello,
I'm not too familiar with Mermaid.js, but after a quick look, wouldn't the [Web Browser Developer Tools](https://developer.chrome.com/docs/devtools/open/) be sufficient to display the ID? I understand you only need a temporary indication for easier connecting, and not for the end user to see after hovering, right?
I'm attaching a video to show what I mean:

<details>
<summary>Video</summary>


https://github.com/squidfunk/mkdocs-material/assets/34622465/521029e6-7461-4309-b367-c74654d330e1



</details>

**Reply by Foxxsayy** [reply]
With that method I either have to 
1. Inspect Element > Hunt through nearby code, often open several drop-downs or
2. Hunt through the code to highlight the element, still frequently opening drop downs.

- (And what I've been doing is CTRL+F finding text in the node and hunting through the text.)

It's a good thought, but still time consuming. Since I need to search the markdown/code for it anyway, I think it might add time onto the search sometimes (unless I'm missing something). It would be awesome if double-clicking a line in the chart brought you to that segment of code or something.

**Comment by alexvoss** [comment]
It might be worthwhile to ask over at https://github.com/orgs/mermaid-js/discussions. IMHO for Material this is an upstream issue, not really something that should be addressed here. We would end up providing extensions/fixes for all sorts of things.
